### PR TITLE
Lists and mermaid

### PR DESCRIPTION
**Pull request by becauselol**
basic implementation for paragraphs and stuff
and basic list without any nesting
will need to fix those in future PRs

### CODE REVIEW

File: src/parser/parser.c [hunk 1]
```diff
@@ -46,107 +84,108 @@ void parse_line(md_node *root, const char *line, int line_length) {
...
-      // printf("\n");
-      // It's really weird why it's not 0 and 13, wasted a lot of time here
-      // Fix it please if you can
-      if (strncmp(line, "```", 3) == 13) {
-          // Exiting the Mermaid block
-          printf("exiting mermaid block\n");
-          in_mermaid_block = 0;
-          printf("%s", mermaid_code);
-          printf("\n");
-          md_node *mermaid_node = create_md_node(NODE_MERMAID_DIAGRAM, mermaid_code, strlen(mermaid_code), NULL, NULL, root, NULL, NULL, MODE_NONE);
```
Comment (kaveripriyap) [inline src/parser/parser.c:99]: where is the `NODE_MERMAID_DIAGRAM` node created if this is deleted? i might have overlooked but i couldnt find it

Comment (becauselol) [reply src/parser/parser.c:99]: changed it so mermaid code is in node_code_block

File: src/node.c [hunk 2]
```diff
@@ -87,7 +93,7 @@ md_node *create_empty_md_node(NodeType type) {
 NodeType md_node_get_type(md_node *node) { return node->type; }
 char *md_node_get_fence_info(md_node *node) { return node->code_language; }
 char *md_node_get_literal(md_node *node) { return node->data; }
-char *md_node_get_mermaid_code(md_node *node) { return node->mermaid_code; }
+char *md_node_get_mermaid_code(md_node *node) { return node->data; }
```
[the remark this answers is not in the document]
Comment (becauselol) [reply src/node.c:96]: fixed and store data in node->data now since its in node_code_block

**Review by kaveripriyap — APPROVED** [review]
LGTM

**Pull request merged**